### PR TITLE
64 speakers etl

### LINKED ISSUE #64: Migrate Person data to Speakers
We need to pull Person data from paper-signs contentful into Speaker content type for CLC app. Please follow below guidance for migration.

---

High-level task breakdown
- update speaker content type to support updates
- import person data from paper-signs into Speaker objects

Speaker Content type changes
- [ ] add email address (string)
- [ ] add team (string)
- [ ] default "on conference directory" boolean to yes
- [ ] if person image (asset) migration is easy then do it, otherwise forego

Speaker import guidance from paper signs Person:
- email maps to email
- `Person.title` should overwrite `Speaker.summary`
- team should be derived from the Team objects that include the Person and where team is "listed" `yes`
- If a Speaker exists with a matching name, update email, team
- If Speaker bio is empty... add the following content:

```
__{insert Person title}__
[Read ___'s Bio](staff bio link on wm.org)
```

### PR DESCRIPTION
**Pull request by gburgett**
Writes an ETL app to load 400+ person records plus profile images to clc

Much of this was copied from the Papyrus ETL app

fixes #64

### CODE REVIEW

File: speakers-etl/src/load/publishesEntries.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      if (!['UnresolvedLinks', 'InvalidEntry'].includes(ex.name)) {
+        throw ex
+      }
+
+      const data = JSON.parse(ex.message)
+      const err = data.details.errors[0]
+      if (!err) {
+        throw ex
+      }
+      if (err.name == 'unique' && err.path.join('.') == 'fields.slug.en-US') {
```
Comment (jpowell) [inline speakers-etl/src/load/publishesEntries.ts:76]: is the logic for this allowing dups to be put in the system, but just remain unpublished?

File: speakers-etl/src/transform.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import { Asset, Entry, Link } from "./contentful/types"
+import { present } from "./util"
+
+export function transformPersonToSpeaker(entry: Entry<PersonProps>, personIdToTeam: Record<string, Entry>): any {
+  const name = [entry.fields.firstName, entry.fields.lastName].filter(present).join(' ')
+  const bio = 
+    entry.fields.hasProfilePage &&
```
Comment (jpowell) [inline speakers-etl/src/transform.ts:7]: 👍🏽

File: speakers-etl/src/transform.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        }
+      }
+    },
+    fields: {
+      internalTitle: { 'en-US': name },
+      name: { 'en-US': name },
+      summary: { 'en-US': entry.fields.title },
+      biography: { 'en-US': bio },
+      isOnConferenceDirectory: { 'en-US': entry.fields.onStaff },
+      team: { 'en-US': team?.fields?.name || entry.fields.team },
```
Comment (jpowell) [inline speakers-etl/src/transform.ts:31]: 👍🏽

**Review by jpowell — COMMENTED** [review]
My only concern is linking to an asset cross space. Is there a potential issue with this? I may be misreading the code.. but that's what it looks like we're doing–just creating a new asset with the same file meta, including URL.

File: speakers-etl/src/load/publishesEntries.ts (hunk 1 above)
Comment (gburgett) [reply speakers-etl/src/load/publishesEntries.ts:76]: I think this is leftover copy-paste, I'll remove it

**Comment by gburgett** [comment]
> My only concern is linking to an asset cross space. Is there a potential issue with this? I may be misreading the code.. but that's what it looks like we're doing–just creating a new asset with the same file meta, including URL.

I'm using the `upload` property, which causes Contentful to pull it from the remote URL and download it to the new space.  I found out about that property by going into the Contentful UI and viewing the network tab as I did a new asset upload

**Comment by jpowell** [comment]
> > My only concern is linking to an asset cross space. Is there a potential issue with this? I may be misreading the code.. but that's what it looks like we're doing–just creating a new asset with the same file meta, including URL.
> 
> I'm using the `upload` property, which causes Contentful to pull it from the remote URL and download it to the new space. I found out about that property by going into the Contentful UI and viewing the network tab as I did a new asset upload

Nice!

**Pull request merged**